modules/ocf/files/ssl/ocf-lets-encrypt

```diff
@@ -0,0 +1,253 @@
+            print('stderr from acme-tiny:', file=sys.stderr)
+            for line in stderr.decode().split('\n'):
+                print('>', line, file=sys.stderr)
+            raise
+    else:
+        return False
+
+
+@contextmanager
+def maybe_start_webserver(dry_run):
+    if not dry_run and socket.socket().connect_ex(('127.0.0.1', 80)) != 0:
+        proc = subprocess.Popen(
+            ('python3', '-m', 'http.server', '80'),
+            cwd=str(ACME_BASE_DIR),
+        )
+
+        # Wait for server to start up before proceeding
+        while socket.socket().connect_ex(('127.0.0.1', 80)) != 0:
+            pass
+
+        yield proc
+
+        proc.kill()
+        print('Stopped ephemeral web server on port 80')
+    else:
+        yield None
+
+
+def main(argv=None):
+    parser = argparse.ArgumentParser(
+        description=__doc__,
+        formatter_class=argparse.RawDescriptionHelpFormatter,
+    )
+    parser.add_argument('-n', '--dry-run', action='store_true')
+    parser.add_argument('-v', '--verbose', action='store_true')
+    parser.add_argument(
+        '--private-key',
+        required=True,
+        help='Path to the private SSL key to use in certificate requests',
+    )
+    parser.add_argument(
+        '--cert-dir',
+        default=DEFAULT_CERT_STORAGE_DIR,
+        help='Where to find and store acquired cert files. '
+             'Defaults to ' + DEFAULT_CERT_STORAGE_DIR,
+    )
+    parser.add_argument(
+        'domains',
+        nargs='+',
+        help='List of domains to acquire/renew certs for',
+    )
+    args = parser.parse_args()
+
+    if args.verbose:
+        global debug
+        debug = functools.partial(print, flush=True)
+
+    requests = set()
+    for domain in args.domains:
+        assert re.match('^[a-z\-_\.0-9]+$', domain)
+        cert_path = Path(args.cert_dir) / '{}.crt'.format(domain)
+
+        if cert_path.exists():
+            expires = expiration_date(cert_path)
+            time_remaining = expires - datetime.now(timezone.utc)
+            if time_remaining > RENEW_WHEN_TIME_REMAINING:
+                debug('not renewing {} (remaining: {})'.format(domain, time_remaining))
+            else:
+                debug('renewing {} (remaining: {})'.format(domain, time_remaining))
+                requests.add(Request(domain, cert_path, has_cert=True))
+        else:
+            requests.add(Request(domain, cert_path, has_cert=False))
+
+    if requests:
+        with maybe_start_webserver(dry_run=args.dry_run):
+            for request in requests:
+                acquire(
+                    request,
+                    key=args.private_key,
+                    dry_run=args.dry_run,
+                )
+
+    need_cert = {request for request in requests if not request.has_cert}
+    debug('Want cert: {}'.format(len(need_cert)))
+    debug('Already have cert: {}'.format(len(args.domains) - len(need_cert)))
+
+    return 255 if requests else 0
```

Comment:
Uh, why's it an error if there are requests?

Also 255 tends to be a special exit code. Try picking another, like 3?

Reply:
See https://docs.google.com/document/d/1teNCynslk71N0UU-O_GiSbCMBLZkAvL8SEot4RN1U5E

Reply:
It would be cool to make `lets-encrypt-update` trigger `build-vhosts` now that I think of it.